lib/utils.js

```diff
@@ -2,22 +2,48 @@
 
 const { request } = require('undici')
 
-async function refreshAccessToken ({ refreshEndpoint, refreshToken, clientId }) {
-  const { statusCode, body } = await request(`${refreshEndpoint}/token`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
+async function refreshAccessToken ({ idpTokenUrl, refreshToken, clientId, clientSecret, contentType }) {
```

Comment:
A typical interface for such action would also accept a `scope: string` and `resource: string[]` options to additionally send to the token endpoint.

Scope is a string composed of individual scope values delimited by a " " (space) to request (in client_credentials) or downscope (in refresh_token).

Resource would typically be a single string but may in fact be multiple string in which case they're sent as two application/x-www-form-urlencoded `resource` parameters.

Auth0 also supports its proprietary `audience` parameter which is effectively `resource` but can only be provided once.